src/routes/user.router.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from 'express';
+import { createValidator } from 'express-joi-validation';
+
+import * as userList from '../data/users.json';
+import UserController from '../controllers/user.controller';
+import { userSchema } from '../validations/user.validation';
+import { userServiceInstance } from '../services/user.service';
+
+const userRouter = Router();
+const validator = createValidator({});
+const userController = new UserController(userServiceInstance);
+
+userRouter.get('/autoSuggest', userController.getAutoSuggestUsers.bind(userController))
+      .get('/:id', userController.getUserById.bind(userController))
+      .get('/', userController.getAllUsers.bind(userController))
+      .post('/', validator.body(userSchema), userController.createUser.bind(userController))
+      .put('/', validator.body(userSchema), userController.updateUser.bind(userController))
```

Comment:
I hope you should update some specific record by `/:id` (take a look at http://expressjs.com/en/4x/api.html#router.param)